tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go

```diff
@@ -89,41 +91,46 @@ func (e *EtcdNodeDeleteAndReplace) TestEtcdNodeDeletionAndReplacement() {
 				machines, err := e.client.Steve.SteveType(machineSteveResourceType).List(query)
 				require.NoError(e.T(), err)
 
-				numOfEtcdNodesBeforeDeletion := 0
-				etcdNodeToDelete := v1.SteveAPIObject{}
-
-				for _, machine := range machines.Data {
-					if machine.Labels[etcdLabel] == "true" && machine.Labels[clusterLabel] == clusterResp.Name {
-						etcdNodeToDelete = machine
-						numOfEtcdNodesBeforeDeletion++
+				previousNodeName := ""
+				for _, nodeLabel := range []string{etcdLabel, workerLabel, controlPlaneLabel} {
+					numNodesBeforeDeletions := 0
+					nodeToDelete := v1.SteveAPIObject{}
+
+					for _, machine := range machines.Data {
+						if machine.Labels[nodeLabel] == "true" && machine.Labels[clusterLabel] == clusterResp.Name {
+							if machine.Name != previousNodeName {
+								nodeToDelete = machine
+							}
+							numNodesBeforeDeletions++
+						}
 					}
-				}
+					previousNodeName = nodeToDelete.Name
+					logrus.Info(fmt.Sprintf("Deleting a %s node: %s ...", nodeLabel, nodeToDelete.Name))
+					err = e.client.Steve.SteveType(machineSteveResourceType).Delete(&nodeToDelete)
+					require.NoError(e.T(), err)
 
-				logrus.Info("Deleting, " + etcdNodeToDelete.Name + " etcd node..")
-				err = e.client.Steve.SteveType(machineSteveResourceType).Delete(&etcdNodeToDelete)
-				require.NoError(e.T(), err)
+					clusterId, err := clusters.GetClusterIDByName(client, clusterResp.Name)
```

Comment:
nit; let's change `clusterId` to be `clusterID` to be more consistent with other tests that we have in the framework.